### ISSUE TITLE
Best practice for defining mapping for inherited models

### ISSUE THREAD

**Issue by k06a** [Yalantis/FastEasyMapping#73]
I just need to create copy of parents `FEMMapping` with different entity name and add some fields. Is there any best practice how to do it without possible problems? Maybe some shortcuts?

```objective-c
+ (FEMMapping *)defaultMapping {
    FEMMapping *superDefaultMapping = [super defaultMapping];
    FEMMapping *mapping = [[FEMMapping alloc] initWithEntityName:NSStringFromClass([self class])];
    for (FEMAttribute *attribute in superDefaultMapping.attributes) {
        [mapping addAttribute:attribute];
    }
    for (FEMRelationship *relationship in superDefaultMapping.relationships) {
        [mapping addRelationship:relationship];
    }
    mapping.primaryKey = superDefaultMapping.primaryKey;
    mapping.rootPath = superDefaultMapping.rootPath;

    //
    
    [mapping addAttributesFromArray:@[
        @"cookie",
    ]];
    return mapping;
}
```

**Comment by dimazen** [comment]
@k06a hello! 
Thats a good question. I can suggest to add a method to the parent that adds all of the attributes and later invoke it from both subclass and a superclass. However it looks ugly. 

Actually, it worth nothing to add conformance of NSCopying, the only blocker at the moment is that `entityName` and `objectClass` are readonly. I'm thinking of conforming to the NSCopying and:
- provide special methods such as `-copyWithEntityName:` and `-copyWithObjectClass:` (looks also weird)
- set `entityName` and `objectClass` to readwrite to allow perform copy and the reassign `entityName` to the required value (as example). 

What do you think?

**Comment by dimazen** [comment]
Just to add how I was doing this before. In the parent I have a method that creates "default" version of mapping and looks like:
```
+ (FEMMapping *)defaultMapping {
    FEMMapping *mapping = [[FEMMapping alloc] initWithEntityName:[self entityName]];
    // add default fields such as Primary key, etc
    return mapping;
}
```

Since I'm using `[self entityName]` it gets correctly picked up by both parent and child. Later you can even override `+defaultMapping` in the subclass and by calling `[super defaultMapping]` you should receive correct mapping with subclasses entity name.

**Comment by k06a** [comment]
Nice idea with `entityName` method overriding. I see no reason to keep entityName and objectClass read only.

**Comment by dimazen** [comment]
@k06a ok, then lets stick to the `NSCopying` and `readwrite` for `entityName` and `objectClass`. I should be able to pick it up on this wednesday :)

**Comment by dimazen** [comment]
Closed by (1.1.1)[https://github.com/Yalantis/FastEasyMapping/releases/tag/1.1.1]
